mlfinlab/portfolio_optimization/mean_variance.py

```diff
@@ -13,52 +16,110 @@ class MeanVarianceOptimisation:
     problem.
     '''
 
-    def __init__(self):
+    def __init__(self, calculate_expected_returns='mean'):
         self.weights = list()
+        self.portfolio_risk = None
+        self.portfolio_return = None
+        self.portfolio_sharpe_ratio = None
+        self.calculate_expected_returns = calculate_expected_returns
+        self.returns_estimator = ReturnsEstimation()
+        self.weight_bounds = None
 
-    def allocate(self, asset_prices, solution='inverse_variance', resample_by='B'):
+    def allocate(self,
+                 asset_names,
+                 asset_prices=None,
+                 expected_asset_returns=None,
+                 covariance_matrix=None,
+                 solution='inverse_variance',
+                 risk_free_rate=0.05,
+                 target_return=0.2,
+                 weight_bounds=(0, 1),
+                 resample_by=None):
+        # pylint: disable=invalid-name, too-many-branches, bad-continuation
         '''
         Calculate the portfolio asset allocations using the method specified.
 
         :param asset_prices: (pd.Dataframe) a dataframe of historical asset prices (daily close)
+        :param expected_asset_returns: (list/np.array/pd.dataframe) a list of mean stock returns (mu)
+        :param covariance_matrix: (pd.Dataframe/numpy matrix) user supplied covariance matrix of asset returns (sigma)
         :param solution: (str) the type of solution/algorithm to use to calculate the weights
```

Comment:
Could the options be listed? E.g. Available algorithms are 'inverse_variance', 'min_volatility', 'max_sharpe'?

Reply:
Added

Reply:
Where? Its not here